asserts/pool.go

```diff
@@ -478,36 +645,65 @@ func (p *Pool) Add(a Assertion, grouping Grouping) (ok bool, err error) {
 }
 
 func (p *Pool) addToGrouping(a Assertion, grouping Grouping, deserializeGrouping func(string) (*internal.Grouping, error)) (ok bool, err error) {
-	uniq := a.Ref().Unique()
-	var u *unresolvedRec
-	var extrag *internal.Grouping
-	var unresolved map[string]*unresolvedRec
-	if u = p.unresolved[uniq]; u != nil {
-		unresolved = p.unresolved
-	} else if u = p.prerequisites[uniq]; u != nil {
-		unresolved = p.prerequisites
-	} else {
-		ok, err := p.isPredefined(a.Ref())
-		if err != nil {
-			return false, err
+	var uniq string
+	ref := a.Ref()
+	// deal with sequence key
+	if ref.Type.SequenceForming() {
+		atseq := AtSequence{
+			Type:        ref.Type,
+			SequenceKey: ref.PrimaryKey[:len(ref.PrimaryKey)-1],
 		}
-		if ok {
-			// nothing to do
-			return true, nil
+		uniq = atseq.Unique()
+	} else {
+		uniq = ref.Unique()
+	}
+	var u unresolvedAssertRecord
+	var extrag *internal.Grouping
+	var unresolved map[string]unresolvedAssertRecord
+
+	if ref.Type.SequenceForming() {
+		if u = p.unresolvedSequences[uniq]; u != nil {
+			unresolved = p.unresolvedSequences
+		} else {
```

Comment:
we probably want a comment here that sequence-forming assertion are never prerequisites so far